src/curl.c

```diff
@@ -437,7 +440,34 @@ static int cc_config_add_page(oconfig_item_t *ci) /* {{{ */
       status = cf_util_get_string(child, &page->plugin_name);
     else if (strcasecmp("URL", child->key) == 0)
       status = cf_util_get_string(child, &page->url);
-    else if (strcasecmp("User", child->key) == 0)
+    else if (strcasecmp("AddressFamily", child->key) == 0) {
+      char *af = NULL;
+      status = cf_util_get_string(child, &af);
+      if (status != 0 || af == NULL) {
+        WARNING("curl plugin: Cannot parse value of `%s' "
+                "for instance `%s'.",
+                child->key, page->instance);
+      } else if (strcasecmp("any", af) == 0) {
+        page->address_family = CURL_IPRESOLVE_WHATEVER;
+      } else if (strcasecmp("ipv4", af) == 0) {
+        page->address_family = CURL_IPRESOLVE_V4;
+      } else if (strcasecmp("ipv6", af) == 0) {
+        /* If curl supports ipv6, use it. If not, log a warning and
+         * fall back to default - don't set status to non-zero.
+         */
+        curl_version_info_data *curl_info = curl_version_info(CURLVERSION_NOW);
+        if (curl_info->features & CURL_VERSION_IPV6)
+          page->address_family = CURL_IPRESOLVE_V6;
+        else
+          WARNING("curl plugin: IPv6 not supported by this libCURL. "
+                  "Using fallback `any'.");
+      } else {
+        WARNING("curl plugin: Unsupported value of `%s' "
```

Comment:
As you set `status = -1` here, I think we have to use `ERROR` instead of `WARNING` here.

Reply:
Actually, I did what most of this plugin's code does, `WARNING` with status `-1` is quite common all over the place
```
      WARNING("curl plugin: Option `%s' not allowed here.", child->key);
      status = -1;
```
```
      WARNING("curl plugin: `Regex' missing in `Match' block.");
      status = -1;
```
```
      WARNING("curl plugin: `Type' missing in `Match' block.");
      status = -1;
```
etc

Reply:
Yes, that is an open question, how to handle configuration errors in Collectd at all.

Ok, since there are many other similar places, that `WARNING` is ok.